addons/website_sale/controllers/main.py

```diff
@@ -563,6 +563,14 @@ def checkout_form_validate(self, mode, all_form_values, data):
             error_message.append(_('Invalid Email! Please enter a valid email address.'))
 
         # vat validation
+        error, error_message = self._vat_validation(data, error, error_message)
+
+        if [err for err in error.values() if err == 'missing']:
+            error_message.append(_('Some required fields are empty.'))
+
+        return error, error_message
+
+    def _vat_validation(self, data, error, error_message):
```

Comment:
Is not it simpler to just extend the fields passed to the new method instead of having to copy the entire method?